doc/man3/EVP_MAC.pod

```diff
@@ -0,0 +1,338 @@
+
+EVP_MAC_final() does the final computation and stores the result in
+the memory pointed at by C<out>, and sets its size in the B<size_t>
+the C<poutlen> points at.  If C<out> is B<NULL>, then no computation
+is made.  To figure out what the output length will be and allocate
+space for it dynamically, simply call with C<out> being B<NULL> and
+C<poutlen> pointing at a valid location, then allocate space and make
+a second call with C<out> pointing at the allocated space.
+
+EVP_MAC_ctrl() is used to manipulate or get information on aspects of
+the MAC which may vary depending on the MAC algorithm or its
+implementation.  This includes the MAC key, and for MACs that use
+other algorithms to do their computation, this is also the way to tell
+it which one to use.  This functions takes variable arguments, the
+exact expected arguments depend on C<cmd>.  EVP_MAC_ctrl() can be
+called both before and after EVP_MAC_init(), but the effect will
+depend on what control is being use.  See </CONTROLS> below for a
+description of standard controls.
+
+EVP_MAC_vctrl() is the variant of EVP_MAC_ctrl() that takes a
+C<va_list> argument instead of variadic arguments.
+
+EVP_MAC_ctrl_str() is an alternative to EVP_MAC_ctrl() to control the
+MAC implementation as E<lt> C<type>, C<value> E<gt> pairs.  The MAC
+implementation documentation should specify what control type strings
+are accepted.
+
+EVP_MAC_str2ctrl() and EVP_MAC_hex2ctrl() are helper functions to
+control the MAC implementation with raw strings or with strings
+containing hexadecimal numbers.  The latter are decoded into
+bitstrings that are sent on to EVP_MAC_ctrl().
+
+=head2 Information functions
+
+EVP_MAC_size() returns the MAC output size for the given context.
+
+EVP_MAC_nid() returns the numeric identity of the given MAC implementation.
+
+EVP_MAC_name() returns the name of the given MAC implementation.
+
+=head2 Object database functions
+
+EVP_get_macbyname() fetches a MAC implementation from the object
+database by name.
+
+EVP_get_macbynid() fetches a MAC implementation from the object
+database by numeric identity.
+
+EVP_get_macbyobj() fetches a MAC implementation from the object
+database by ASN.1 OBJECT (i.e. an encoded OID).
+
+=head1 CONTROLS
+
+The standard controls are:
+
+=over 4
+
+=item B<EVP_MAC_CTRL_SET_KEY>
+
+This control expects two arguments: C<unsigned char *key>, C<size_t keylen>
+
+These will set the MAC key from the given string of the given length.
+The string may be any bitstring, and can contain NUL bytes.
+
+For MACs that use an underlying computation algorithm, the algorithm
+I<must> be set first, see B<EVP_MAC_CTRL_SET_ENGINE>,
+B<EVP_MAC_CTRL_SET_MD> and B<EVP_MAC_CTRL_SET_CIPHER> below.
+
+=item B<EVP_MAC_CTRL_SET_FLAGS>
+
+This control expects one arguments: C<unsigned long flags>
+
+These will set the MAC flags to the given numbers.  Some MACs do not
+support this option.
+
+=item B<EVP_MAC_CTRL_SET_ENGINE>
```

Comment:
Do we require this given providers are the future...

Reply:
When (and if, there's still an "if") we get there, this will need to be redone anyway...  This PR deals with "now".